src/lib/mina_lib/mina_lib.ml

```diff
@@ -556,16 +562,46 @@ let create_sync_status_observer ~logger ~is_seed ~demo_mode ~net
       | None ->
           ()
     in
-    let handle_status_change status =
-      if match status with `Offline -> true | _ -> false then
-        start_offline_timeout ()
-      else stop_offline_timeout ()
+    let log_bootstrap_error_and_restart _tm =
+      [%log error] "Daemon has been in bootstrap for %0.0f minutes, restarting"
+        bootstrap_timeout_min ;
+      raise Bootstrap_stuck_shutdown
+    in
+    let start_bootstrap_timeout () =
+      match !bootstrap_timeout with
+      | Some _ ->
+          ()
+      | None ->
+          bootstrap_timeout :=
+            Some
+              (Timeout.create () bootstrap_timeout_duration
+                 ~f:log_bootstrap_error_and_restart )
+    in
+    let stop_bootstrap_timeout () =
+      match !bootstrap_timeout with
+      | Some timeout ->
+          Timeout.cancel () timeout () ;
+          bootstrap_timeout := None
+      | None ->
+          ()
+    in
+    let handle_status_change state status =
```

Comment:
Doesn't `status` here include `Bootstrap` based on how `incremental_status` is defined [here](https://github.com/MinaProtocol/mina/blob/45e909764f645f85cc88cb9a465367b3e1a5a10f/src/lib/mina_lib/mina_lib.ml#L462) ?  How is `state` tracked?

Reply:
Yes, I got fooled by the type that merlin was reporting. We only need `status`.